src/prolong_restrict/pr_loops.hpp

```diff
@@ -130,16 +130,28 @@ ProlongationRestrictionLoop(const ProResInfoArr_t &info, const Idx_t &buffer_idx
         const std::size_t buf = buffer_idxs(sub_idx);
         if (DoRefinementOp(info(buf), op)) {
           using TE = TopologicalElement;
-          if (info(buf).fine.topological_type == TopologicalType::Cell)
+          if (info(buf).include_el[static_cast<int>(TE::CC)])
             IterateInnerProlongationRestrictionLoop<DIM, Stencil, TE::CC>(
                 team_member, buf, info, ckb, cjb, cib, kb, jb, ib);
-          if (info(buf).fine.topological_type == TopologicalType::Face)
-            IterateInnerProlongationRestrictionLoop<DIM, Stencil, TE::F1, TE::F2, TE::F3>(
+          if (info(buf).include_el[static_cast<int>(TE::F1)])
+            IterateInnerProlongationRestrictionLoop<DIM, Stencil, TE::F1>(
                 team_member, buf, info, ckb, cjb, cib, kb, jb, ib);
-          if (info(buf).fine.topological_type == TopologicalType::Edge)
-            IterateInnerProlongationRestrictionLoop<DIM, Stencil, TE::E3, TE::E2, TE::E1>(
+          if (info(buf).include_el[static_cast<int>(TE::F2)])
+            IterateInnerProlongationRestrictionLoop<DIM, Stencil, TE::F2>(
                 team_member, buf, info, ckb, cjb, cib, kb, jb, ib);
-          if (info(buf).fine.topological_type == TopologicalType::Node)
+          if (info(buf).include_el[static_cast<int>(TE::F3)])
```

Comment:
Maybe we should make a method in `BufferInfo` that takes a topological type enum so we don't have to do this static cast "downstream"? might be easier to read.

Reply:
done